### PR TITLE
Always shuffle data in default automl data split strategies

### LINKED ISSUE #1259: Poor performance on diamond dataset
**Problem**
Automl yields models with negative R2.

**Repro**
Dataset [here](https://alteryx0-my.sharepoint.com/:x:/g/personal/dylan_sherry_alteryx_com/EVJqRYjya5JPmr-xlHDEM08BC4cdo4wgP_WfVubuoGpgaw?e=HeMaw4).
```python
import evalml
import pandas as pd
import numpy as np
df = pd.read_csv('stones_encoded_small.csv')
y = df.pop('total_sales_price')
automl = evalml.automl.AutoMLSearch(problem_type='regression')
automl.search(df, y)
```
Data checks will fail due to highly null / single-value columns. You can disable them with `data_checks='disabled'`. Or, to address them and continue:
```python
cols_to_drop = ['culet_condition', 'fancy_color_dominant_color', 'fancy_color_intensity', 'fancy_color_overtone', 'fancy_color_secondary_color', 'fluor_color', 'image_file_url', 'diamond_id', 'currency_code', 'currency_symbol', 'fancy_color_dominant_color', 'fancy_color_intensity', 'fancy_color_overtone', 'fancy_color_secondary_color', 'has_sarineloupe']
df.drop(columns=cols_to_drop, inplace=True)
automl = evalml.automl.AutoMLSearch(problem_type='regression')
automl.search(df, y)
```
The results are highly similar either way: negative R2 values for all models, i.e. the models can't produce meaningful results.

Switching the metric to MSE and MAE yields similarly poor models.

**Discussion**
My first suspicion is that the features aren't getting the right type. When I look at the dtypes inferred by pandas, I see many are set as `float64` but only have a few unique values, i.e. they should be set as categorical. I gave that a shot but it didn't seem to change the model results, so there's more to the story.

### PR DESCRIPTION
**Pull request by dsherry**
Fix #1259

### CODE REVIEW

File: evalml/tests/automl_tests/test_automl.py [hunk 1]
```diff
@@ -645,6 +645,32 @@ def generate_fake_dataset(rows):
...
+    # increases monotonically from 0 to n-1.
+    #
+    # if shuffle is enabled, the baseline model, which predicts the mean of the training data, should accurately
+    # predict the mean of the validation data, because the training split in each CV fold will contain a mix of
+    # values from across the target range, thus yielding an R^2 of close to 0.
+    #
+    # if shuffle is disabled, the mean value learned on each CV fold's training data will be incredible inaccurate,
+    # thus yielding an R^2 well below 0.
+
+    n = 100000
```
Comment (freddyaboulton) [inline evalml/tests/automl_tests/test_automl.py:659]: If this test doesn't take too long it might be worth running it with n=1000 so that the KFold CV is used?

**Review by freddyaboulton — APPROVED** [review]
@dsherry This looks good to me!

File: evalml/tests/automl_tests/test_automl.py [hunk 2]
```diff
@@ -645,6 +645,32 @@ def generate_fake_dataset(rows):
     assert automl.data_split.test_size == (automl._LARGE_DATA_PERCENT_VALIDATION)
 
 
+def test_data_split_shuffle():
```
Comment (jeremyliweishih) [inline evalml/tests/automl_tests/test_automl.py:648]: great test!

**Review by jeremyliweishih — APPROVED** [review]
LGTM

File: evalml/automl/data_splitters/training_validation_split.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
 class TrainingValidationSplit(BaseCrossValidator):
     """Split the training data into training and validation sets"""
 
-    def __init__(self, test_size=None, train_size=None, shuffle=True, stratify=None, random_state=0):
+    def __init__(self, test_size=None, train_size=None, shuffle=False, stratify=None, random_state=0):
```
Comment (gsheni) [inline evalml/automl/data_splitters/training_validation_split.py:8]: Should this be `shuffle=True`?

**Review by angela97lin — APPROVED** [review]
Great, LGTM!

File: evalml/automl/data_splitters/training_validation_split.py (hunk 3 above)
Comment (dsherry) [reply evalml/automl/data_splitters/training_validation_split.py:8]: @gsheni I updated this to be the same as the other sklearn-defined CV methods, which have `shuffle=False` by default. Then, in the automl code which calls this, I set `shuffle=True`. I just did it to be consistent with sklearn.

File: evalml/tests/automl_tests/test_automl.py (hunk 1 above)
Comment (dsherry) [reply evalml/tests/automl_tests/test_automl.py:659]: Ah shoot you're right. I'll track this down. I upped the number because I wasn't reproing it as reliably at lower values.
